lib/listener/exit.js

```diff
@@ -24,6 +32,12 @@ module.exports = function () {
     }
   });
 
+  event.dispatcher.on(event.all.after, () => {
+    // Writes the Failed Test Names In The JSON File For Rerun
+    failedTestRerun.writeFailedTests(failedTestName);
```

Comment:
this is not a correct place to put this logic
this listener only sets the exit code for failed tests